router/src/infer.rs

```diff
@@ -198,6 +198,7 @@ impl Infer {
                 messages,
                 eos_token: eos_token.as_deref(),
                 bos_token: bos_token.as_deref(),
+                add_generation_prompt: true,
```

Comment:
Why is it sometimes `true`, sometimes `false`?

Reply:
the `true` values will be used in all cases at runtime. The `false`s are only used in the existing template tests (and those tests all do not have `add_generation_prompt` so both true and false will return the same results).

In terms of updating the tests/add a new one I can push to this PR or maybe follow up with another small PR to avoid this blocking anything. lmk whats best! thank you!

Reply:
***update:

pushed new test and updated old ones